src/test/java/uk/gov/hmcts/reform/ccd/document/am/controller/advice/CaseDocumentControllerAdviceTest.java

```diff
@@ -0,0 +1,54 @@
+package uk.gov.hmcts.reform.ccd.document.am.controller.advice;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.springframework.http.converter.HttpMessageConversionException;
+import uk.gov.hmcts.reform.ccd.document.am.controller.WelcomeController;
+import uk.gov.hmcts.reform.ccd.document.am.controller.advice.exception.*;
+
+class CaseDocumentControllerAdviceTest {
+
+    private transient WelcomeController welcomeController = new WelcomeController();
+
+    @Test
+    void testHandleRequiredFieldMissingException() {
+        Assertions.assertThrows(RequiredFieldMissingException.class, () -> {
+            welcomeController.getException("requiredFieldMissingException");
+        });
+    }
+
+    @Test
+    void testInvalidRequest() {
+        Assertions.assertThrows(InvalidRequest.class, () -> {
+            welcomeController.getException("invalidRequest");
+        });
+    }
+
+    @Test
+    void testResourceNotFoundException() {
+        Assertions.assertThrows(ResourceNotFoundException.class, () -> {
+            welcomeController.getException("resourceNotFoundException");
+        });
+    }
+
+    @Test
+    void testHttpMessageConversionException() {
+        Assertions.assertThrows(HttpMessageConversionException.class, () -> {
+            welcomeController.getException("httpMessageConversionException");
+        });
+    }
+
+    @Test
+    void testBadRequestException() {
+        Assertions.assertThrows(BadRequestException.class, () -> {
+            welcomeController.getException("badRequestException");
+        });
+    }
+
+    @Test
+    void testCaseNotFoundException() {
+        Assertions.assertThrows(CaseNotFoundException.class, () -> {
+            welcomeController.getException("caseNotFoundException");
+        });
+    }
```

Comment:
is it possible to add some more test like 
1> Type mismatch exception 
2> Boundary value exception etc

Reply:
As discussed this is a generic class for handling exceptions and would be updated gradually based on the new scenarios.